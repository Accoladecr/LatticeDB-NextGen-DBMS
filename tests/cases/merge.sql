CREATE TABLE people (
                        id TEXT PRIMARY KEY,
                        name TEXT MERGE lww,
                        tags SET<TEXT> MERGE gset,
                        credits INT MERGE sum_bounded(0, 1000000)
);
INSERT INTO people (id, name, tags, credits) VALUES
                                                 ('u1','Ada', {'engineer','math'}, 10),
                                                 ('u2','Grace', {'engineer'}, 20);
INSERT INTO people (id, credits, tags, name) VALUES
    ('u1', 15, {'leader'}, 'Ada Lovelace') ON CONFLICT MERGE;
SELECT id, name, credits, tags FROM people;
EXIT;
