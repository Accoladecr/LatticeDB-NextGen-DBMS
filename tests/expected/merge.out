CREATE TABLE
INSERT 2 row(s)
INSERT 1 row(s)
id | name | credits | tags
'u1' | 'Ada Lovelace' | 25 | {'engineer','leader','math'}
'u2' | 'Grace' | 20 | {'engineer'}
Bye.
